src/Conversion/ONNXToTorch/NN/FlattenOp.cpp

```diff
@@ -0,0 +1,175 @@
+ *    In torch side, Calculate Start dim and End dim using this 
+ *    axis attribute value.
+ *
+ * Operands:
+ *    input     tensor of 8-bit/16-bit/32-bit unsigned integer values or
+ *    		tensor of 64-bit unsigned integer values or
+ *    	       	tensor of 8-bit/16-bit/32-bit/64-bit signless integer 
+ *    	       	values or
+ *    		tensor of bfloat16 type or tensor of 16-bit float values
+ *    		tensor of 32-bit float or tensor of 64-bit float values
+ *    		tensor of string type values or tensor of 1-bit signless 
+ *    		integer values or tensor of complex type with 32-bit float
+ *    		elements values or tensor of complex type with 64-bit float
+ *    		elements values or memref of any type values
+ *    Map this input operand into input parameter in torch side.
+ * Results:
+ *    output     tensor of 8-bit/16-bit/32-bit unsigned integer values or
+ *              tensor of 64-bit unsigned integer values or
+ *              tensor of 8-bit/16-bit/32-bit/64-bit signless integer 
+ *              values or
+ *              tensor of bfloat16 type or tensor of 16-bit float values
+ *              tensor of 32-bit float or tensor of 64-bit float values
+ *              tensor of string type values or tensor of 1-bit signless
+ *              integer values or tensor of complex type with 32-bit float
+ *              elements values or tensor of complex type with 64-bit float
+ *              elements values or memref of any type values
+ *
+ */
+
+class ONNXFlattenOpToTorchLowering : public ConversionPattern {
+public:
+  ONNXFlattenOpToTorchLowering(TypeConverter &typeConverter,
+		  MLIRContext *ctx)
+      : ConversionPattern( typeConverter,
+		      ::mlir::ONNXFlattenOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+
+    Location loc = op->getLoc();
+    mlir::MLIRContext *context = op->getContext();
+    ONNXFlattenOp op1 = llvm::dyn_cast<ONNXFlattenOp>(op);
+    ONNXFlattenOpAdaptor adaptor(operands);
+
+    Value input = op1.input();
+    auto axisValue = op1.axis();       // ::mlir::IntegerAttr
+
+    auto inputShape = input.getType().cast<ShapedType>().getShape();
+    int64_t inputRank = inputShape.size();
+
+    TensorType op_tensor_type =
+	    op->getResult(0).getType().cast<TensorType>();
+    auto resultTy = Torch::ValueTensorType::get(op1.getContext(),
+               op_tensor_type.getShape(), op_tensor_type.getElementType());
+
+    TensorType input_tensor_type  = input.getType().cast<TensorType>();
+    auto inputTy =
+	 Torch::ValueTensorType::get(context, input_tensor_type.getShape(),
+                    input_tensor_type.getElementType());
+    auto itt  =
+	  rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(loc,
+                    inputTy, input);
+    auto ty = IntegerType::get(op1.getContext(), 64);
+    int64_t startDim;
+    int64_t endDim;
+    Value atenflaten1 = itt;
```

Comment:
I believe this function would be more readable if fewer intermediate variables were in scope.
Consider writing it like:
```c++
Value result = itt;
if (axisValue != 0) {
   // this function would factorize lines 134-145
   result = createAtenFlattenOp(rewriter, loc, result, resultTy, 0, axisValue - 1);
}
result = createAtenFlattenOp(rewriter, loc, result, resultTy, axisValue, inputRank);
rewriter.replaceOpWithNewOp<...>(..., result);
```
Where the `result` variables stands for `atenflaten1`, `atenflatten2`, and `result` in the current code

Reply:
Resolved